In [3]:
import pandas as pd

In [4]:
amr = pd.read_excel("AMR DATA 7.7.2026.xlsx")

In [5]:
amr.shape

(49302, 60)

In [6]:
amr.describe()

,Macro name,collection year,Specimen type (Numeric),GEH
count,0.0,49246.0,49225.000000,706.000000
mean,NaN,2024.0,12.074901,17.960340
std,NaN,0.0,3.821506,8.663193
min,NaN,2024.0,2.000000,6.000000
25%,NaN,2024.0,11.000000,6.000000
50%,NaN,2024.0,12.000000,20.000000
75%,NaN,2024.0,12.000000,24.000000
max,NaN,2024.0,330.000000,46.000000


In [11]:
amr.info()

<class 'pandas.DataFrame'>
RangeIndex: 49302 entries, 0 to 49301
Data columns (total 60 columns):
 #   Column                            Non-Null Count  Dtype  
---  ------                            --------------  -----  
 0   Macro name                        0 non-null      float64
 1   Country                           49246 non-null  str    
 2   Hospital Name                     49246 non-null  str    
 3   Facility lebel                    49246 non-null  str    
 4   Origin                            30900 non-null  str    
 5   Identification number             49223 non-null  object 
 6   Last name                         15710 non-null  object 
 7   First name                        15708 non-null  object 
 8   Sex                               49209 non-null  object 
 9   Date of birth                     57 non-null     object 
 10  Age                               49163 non-null  object 
 11  Age category                      48181 non-null  object 
 12  Ward           

In [8]:
amr

,Macro name,Country,Hospital Name,Facility lebel,Origin,Identification number,Last name,First name,Sex,Date of birth,...,GEH,Gentamicin,Imipenem,Meropenem,Nitrofurantoin,PEN,Piperacillin/Tazobactam,Tetracycline,Trimethoprim/Sulfamethoxazole,VAN
0,NaN,TZA,MZRH,Zonal hospital,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
1,NaN,TZA,MZRH,Zonal hospital,NaN,01 11 69 40,NaN,NaN,m,NaN,...,NaN,NaN,NaN,NaN,NaN,R,NaN,NaN,R,NaN
2,NaN,TZA,MZRH,Zonal hospital,NaN,1018535,NaN,NaN,m,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3,NaN,TZA,MZRH,Zonal hospital,NaN,1103584,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
4,NaN,TZA,MZRH,Zonal hospital,NaN,1115940,NaN,NaN,f,NaN,...,NaN,18,NaN,29,25,NaN,NaN,NaN,6,NaN
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
49297,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,Intermediate,NaN,NaN,NaN,NaN,NaN,NaN
49298,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,Intermediate,NaN,NaN,NaN,NaN,NaN,NaN
49299,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,Intermediate,NaN,NaN,NaN,NaN,NaN,NaN
49300,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,...,NaN,NaN,NaN,Intermediate,NaN,NaN,NaN,NaN,NaN,NaN


In [9]:
amr.columns

Index(['Macro name', 'Country', 'Hospital Name', 'Facility lebel', 'Origin',
       'Identification number', 'Last name', 'First name', 'Sex',
       'Date of birth', 'Age', 'Age category', 'Ward', 'Institution',
       'Department', 'Location type', 'Date of admission', 'collection year',
       'Specimen number', 'Specimen date', 'Specimen type-Code',
       'Specimen type', 'Specimen type (Numeric)', 'Reason', 'WHO-Pathogens',
       'pathogen_species', 'Gram Stain_Status', 'Organism type', 'Serotype',
       'Beta-lactamase', 'ESBL', 'Carbapenem resistance', 'MRSA',
       'Inducible clindamycin resistance', 'Date of data entry', 'Amikacin',
       'Amoxicillin/Clavulanic acid', 'Ampicillin', 'AZM', 'Cefazolin',
       'Cefepime', 'Cefixime', 'Cefotaxime', 'Cefoxitin', 'Ceftazidime',
       'Ceftriaxone', 'Chloramphenicol', 'Ciprofloxacin', 'Clindamycin',
       'Erythromycin', 'GEH', 'Gentamicin', 'Imipenem', 'Meropenem',
       'Nitrofurantoin', 'PEN', 'Piperacillin/Tazobactam', 

In [12]:
missing = amr.isnull().sum()
missing

Macro name                          49302
Country                                56
Hospital Name                          56
Facility lebel                         56
Origin                              18402
Identification number                  79
Last name                           33592
First name                          33594
Sex                                    93
Date of birth                       49245
Age                                   139
Age category                         1121
Ward                                   93
Institution                          4019
Department                             75
Location type                          63
Date of admission                   31858
collection year                        56
Specimen number                       131
Specimen date                         120
Specimen type-Code                     77
Specimen type                         281
Specimen type (Numeric)                77
Reason                            

In [13]:
missing[missing > 0]

Macro name                          49302
Country                                56
Hospital Name                          56
Facility lebel                         56
Origin                              18402
Identification number                  79
Last name                           33592
First name                          33594
Sex                                    93
Date of birth                       49245
Age                                   139
Age category                         1121
Ward                                   93
Institution                          4019
Department                             75
Location type                          63
Date of admission                   31858
collection year                        56
Specimen number                       131
Specimen date                         120
Specimen type-Code                     77
Specimen type                         281
Specimen type (Numeric)                77
Reason                            

In [14]:
missing_report = pd.DataFrame({
    "Missing Values": amr.isnull().sum(),
    "Percentage": round(amr.isnull().mean()*100,2)
})

missing_report

,Missing Values,Percentage
Macro name,49302,100.00
Country,56,0.11
Hospital Name,56,0.11
Facility lebel,56,0.11
Origin,18402,37.33
Identification number,79,0.16
Last name,33592,68.14
First name,33594,68.14
Sex,93,0.19
Date of birth,49245,99.88


In [15]:
missing_report[missing_report["Missing Values"] > 0]

,Missing Values,Percentage
Macro name,49302,100.00
Country,56,0.11
Hospital Name,56,0.11
Facility lebel,56,0.11
Origin,18402,37.33
Identification number,79,0.16
Last name,33592,68.14
First name,33594,68.14
Sex,93,0.19
Date of birth,49245,99.88


In [21]:
amr['Macro name'].value_counts(dropna=False)

Macro name
NaN    49302
Name: count, dtype: int64

In [19]:
amr['Country'].value_counts(dropna=False)

Country
TZA    49246
NaN       56
Name: count, dtype: int64

In [23]:
amr['Hospital Name'].value_counts(dropna=False)

Hospital Name
BMC           14475
MNH           13579
KCMC           8158
BMHL           5523
MZRH           2897
MoroRRH        1518
TMK            1210
Ligula RRH      974
KGM             912
NaN              56
Name: count, dtype: int64

In [24]:
amr['Facility lebel'].value_counts(dropna=False)

Facility lebel
Zonal hospital       44632
Regional hospital     4614
NaN                     56
Name: count, dtype: int64

In [25]:
amr['Origin'].value_counts(dropna=False)

Origin
h      30900
NaN    18402
Name: count, dtype: int64

In [26]:
amr['Identification number'].value_counts(dropna=False)

Identification number
DIAGNOSTIC    261
NaN            79
M02-19-67      30
M76-68-75      20
M75-71-69      20
             ... 
tmb1399251      1
u213808         1
y338467         1
2-255332        1
2-282286        1
Name: count, Length: 31978, dtype: int64

In [34]:
amr['Sex'].value_counts(dropna=False)

Sex
f      24846
m      24360
NaN       93
6          1
d          1
1          1
Name: count, dtype: int64

In [36]:
amr['pathogen_species'].value_counts(dropna=False)

pathogen_species
No growth                                      32204
Staphylococcus, coagulase negative              4372
Escherichia coli                                2361
Klebsiella pneumoniae ss. pneumoniae            1296
Staphylococcus aureus ss. aureus                1293
                                               ...  
Streptococcus gallolyticus ss. pasteurianus        1
Serratia rubidaea                                  1
Streptococcus sanguinis                            1
Candidatus Xenohaliotis sp.                        1
Yersinia frederiksenii                             1
Name: count, Length: 131, dtype: int64

OBERVATION ~ INSPECTION:

1. Dataset contain 49302 rows and 60 columns
2. Columns data type float64(4), object(40), str(16)
3. Numeric = Macro name, Specimen type (Numeric),  collection year , GEH
4. Obtainded Dataframe for missing values = missing-report.
5. Understand each colums, except = origin h .? other medical terminology...
6. there some duplicates = 249 rows


In [29]:
amr.duplicated().sum()

np.int64(249)

In [31]:
missing_report.to_csv("Missing Report Raw.csv")

In [37]:
amr['pathogen_species'].unique()

<StringArray>
[                                            nan,
                                     'No growth',
                       'Acinetobacter baumannii',
 'Acinetobacter calcoaceticus-baumannii complex',
                             'Acinetobacter sp.',
                                'Aerococcus sp.',
                          'Aeromonas hydrophila',
                                 'Aeromonas sp.',
                         'Acinetobacter lwoffii',
                        'Arthroderma persicolor',
 ...
                      'Streptococcus salivarius',
                       'Streptococcus sanguinis',
                            'Staphylococcus sp.',
                             'Streptococcus sp.',
            'Streptococcus viridans, alpha-hem.',
                   'Candidatus Xenohaliotis sp.',
                            'No pathogens found',
                   'No significant growth found',
                                         'Yeast',
                        'Yersin